app/src/main/java/io/appium/settings/LocationService.java

```diff
@@ -17,20 +17,58 @@
 package io.appium.settings;
 
 import android.app.Service;
+import android.content.Context;
 import android.content.Intent;
 import android.content.pm.PackageManager;
+import android.location.Criteria;
+import android.location.Location;
 import android.location.LocationManager;
+import android.location.LocationProvider;
 import android.os.IBinder;
 import android.util.Log;
 
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.GoogleApiAvailability;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationServices;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import io.appium.settings.location.FusedLocationProvider;
+import io.appium.settings.location.LocationFactory;
+import io.appium.settings.location.LocationManagerProvider;
+import io.appium.settings.location.MockLocationProvider;
+
 public class LocationService extends Service {
     private static final String TAG = "MOCKED LOCATION SERVICE";
 
+    private static final String LONGITUDE_PARAMETER_KEY = "longitude";
+    private static final String LATITUDE_PARAMETER_KEY = "latitude";
+    private static final String ALTITUDE_PARAMETER_KEY = "altitude";
+
+    private static final long UPDATE_INTERVAL_MS = 2000L;
+
+    private LocationFactory locationFactory = new LocationFactory();
+    private Timer locationUpdatesTimer = new Timer();
+    private TimerTask locationUpdateTask;
+    private List<MockLocationProvider> mockLocationProviders;
```

Comment:
I'd rather have this variable already assigned to an empty list and use clear/addAll in onCreate